### ISSUE TITLE
List of tests common to time series analysis

### ISSUE THREAD

**Issue by milktrader** [JuliaStats/HypothesisTests.jl#16]
- [x] Augmented Dickey-Fuller (unit root)
- [ ] Phillips-Perron (unit root)
- [ ] KPSS (unit root)
- [ ] Shapiro-Wilks (normality)
- [x] Jaques-Berra (3rd and 4th moments for normality) (TimeModels has this implemented)
- [x] Ljung-Box (auto correlation)
- [x] Durbin-Watson (auto correlation)
- [x] Breusch-Godfrey (serial correlation)

**Comment by rawls238** [comment]
I'm going to work on implementing these - has any work been done on this?

**Comment by tomaskrehlik** [comment]
Not really.

**Comment by milktrader** [comment]
@rawls238 cool, go for it!

**Comment by BenjaminBorn** [comment]
I have started implementing autocorrelation tests in #87.

**Comment by BenjaminBorn** [comment]
Ok, ADF test is here #92. Please feel free to comment.

@milktrader Should we move the Jarque-Berra test to HypothesisTests as well? It seems a bit lost in TimeModels if all other tests are here.

**Comment by milktrader** [comment]
@BenjaminBorn yes, I think so. TimeModels has not been developed for some time now.

**Comment by ararslan** [comment]
> TimeModels has not been developed for some time now.

In particular, I don't think it even works on Julia versions later than 0.4. If it won't be maintained going forward, perhaps it should be deprecated and the relevant pieces moved to other packages as appropriate? @BenjaminBorn has already been doing excellent work implementing the tests in HypothesisTests; it would just be a matter of finding appropriate homes for the other things.

**Comment by milktrader** [comment]
I agree @ararslan. I could move the package back to my repos and let it sit there until I have resources to develop it.

**Comment by BenjaminBorn** [comment]
I have added the Jarque-Bera test to HypothesisTests (see #95).

Concerning TimeModels, it seems it consists of two main parts. ARIMA modelling based on a state-space approach and some GARCH functionally (transferred from [GARCH.jl](https://github.com/AndreyKolev/GARCH.jl)). Would it make sense to split it again? There are also ARIMA.jl and RARIMA.jl (an R wrapper) but they also seem to be dormant.

**Comment by milktrader** [comment]
Yes, I think a big reason for dormant activity is because it's become too unwieldy.

**Comment by BenjaminBorn** [comment]
Durbin-Watson test added in #102.

**Comment by azev77** [comment]
@pedrochaim has [UnitRoots](https://github.com/pedrochaim/UnitRoot.jl).jl with DFGLS & KPSS.

**Comment by azev77** [comment]
@tomaskrehlik you have some nice code for the [Johansen test for cointegration](https://github.com/tomaskrehlik/VARmodels.jl/blob/master/src/cointegrationJohansen.jl). 
Would it be possible to port it here?

**Comment by azev77** [comment]
[ARCHModels](https://github.com/s-broda/ARCHModels.jl/blob/cf95a9c196428bc8e3ce108b631e43f1a7da6698/src/tests.jl).jl has several time series tests.

**Comment by tomaskrehlik** [comment]
> @tomaskrehlik you have some nice code for the [Johansen test for cointegration](https://github.com/tomaskrehlik/VARmodels.jl/blob/master/src/cointegrationJohansen.jl).
> Would it be possible to port it here?

I haven't really done any Julia in a while but I will put it on my agenda and see when it fits in. :)
